app/src/main/java/com/github/se/polyfit/data/remote/firebase/PostFirebaseRepository.kt

```diff
@@ -1,19 +1,39 @@
 package com.github.se.polyfit.data.remote.firebase
 
 import android.util.Log
+import com.firebase.geofire.GeoFire
+import com.firebase.geofire.GeoLocation
+import com.firebase.geofire.GeoQueryEventListener
 import com.github.se.polyfit.model.post.Post
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.FirebaseDatabase
+import com.google.firebase.firestore.CollectionReference
 import com.google.firebase.firestore.DocumentReference
+import com.google.firebase.firestore.FieldPath
 import com.google.firebase.firestore.FirebaseFirestore
 import com.google.firebase.storage.FirebaseStorage
+import com.google.firebase.storage.StorageReference
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.launch
 import kotlinx.coroutines.tasks.await
+import kotlinx.coroutines.withContext
 
 class PostFirebaseRepository(
     private val db: FirebaseFirestore = FirebaseFirestore.getInstance(),
-    private val pictureDb: FirebaseStorage = FirebaseStorage.getInstance()
+    private val pictureDb: FirebaseStorage = FirebaseStorage.getInstance(),
+    private val rtdb: FirebaseDatabase =
+        FirebaseDatabase.getInstance(
+            "https://polyfit-316e8-default-rtdb.europe-west1.firebasedatabase.app/")
```

Comment:
I know it's not a good practice, but it becomes really tricky to tests, I'll find a better way later

Reply:
We can add that the pile of technical dept that we currently have (lol).
I had to do something very similar with Spoonacular API. I had to have the CI inject the API keys in the gradle.properties. Feel free to hit me up if you have any questions!